src/layouts/Login/components/OtpForm.tsx

```diff
@@ -0,0 +1,100 @@
+import { HStack, FormControl, FormErrorMessage } from "@chakra-ui/react"
+import { Button, FormLabel, Input } from "@opengovsg/design-system-react"
+import { useCallback, useEffect, useState } from "react"
+import { useForm, FormProvider } from "react-hook-form"
+
+export type OtpFormInputs = {
+  otp: string
+}
+
+interface OtpFormProps {
+  email: string
+  onSubmit: (inputs: OtpFormInputs) => Promise<void>
+  onResendOtp: () => Promise<void>
+}
+
+const OTP_TIMER_INTERVAL = 60
+
+export const OtpForm = ({
+  email,
+  onSubmit,
+  onResendOtp,
+}: OtpFormProps): JSX.Element => {
+  const [timer, setTimer] = useState(OTP_TIMER_INTERVAL)
+  const methods = useForm<OtpFormInputs>({
+    mode: "onBlur",
+  })
+  const { handleSubmit, register, formState, setError } = methods
+
+  const validateOtp = useCallback(
+    (value: string) => value.length === 6 || "Please enter a 6 digit OTP.",
+    []
+  )
+
+  const onSubmitForm = async (inputs: OtpFormInputs) => {
+    return onSubmit(inputs).catch((err) => {
+      const {
+        error: { message },
+      } = err.response.data
+      setError("otp", {
+        type: "server",
+        message,
+      })
+    })
+  }
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (timer > 0) setTimer(timer - 1)
+    }, 1000)
+    return () => clearInterval(interval)
+  }, [timer])
+
+  return (
+    <FormProvider {...methods}>
+      <form onSubmit={handleSubmit(onSubmitForm)}>
+        <FormControl
+          isInvalid={!!formState.errors.otp}
+          isReadOnly={formState.isSubmitting}
+        >
+          <FormLabel isRequired htmlFor="otp">
+            {`Enter OTP sent to ${email}`}
+          </FormLabel>
+          <Input
```

Comment:
Doesn't seem to play too well with the existing form stuff - opting to leave it as is especially since Input has a design system equivalent while NumberInput doesn't